app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,57 @@
+class ArticlesController < ApplicationController
+  before_action :find_article, except: %i[index new create]
+  http_basic_authenticate_with name: 'user', password: 'secret', except: %i[index show]
+
+  def index
+    @articles = Article.all
+  end
+
+  def show; end
+
+  def new
+    @article = Article.new
+  end
+
+  def create
+    @article = Article.new(article_params)
+    begin
+      @article.save!
+    rescue ActiveRecord::RecordInvalid
+      puts e.message
+      render :new
+    end
+    redirect_to @article
```

Comment:
This should be inside the `begin` block, like this:

```ruby
    begin
      @article.save!
      redirect_to @article
    rescue ActiveRecord::RecordInvalid
```

Reply:
Done ! and thanks for the recommendation

Reply:
Update of `save` method: [Update Save Method](https://github.com/SanGeeky/rails-training/pull/1#discussion_r612730149)